### PR TITLE
fix JSONSchemaKey generation in struct definition

### PR DESCRIPTION
**Pull request by denis90**
## Overview

- [x] correctly generate struct definition when using JSONSchemaKey

## Notes

Refs: LM-zxy

### CODE REVIEW

File: generator/template.go [hunk 1]
```diff
@@ -293,7 +293,11 @@ func (td *TemplateData) GenSubSchemas() string {
 				buf.WriteString("*kallax.JSONSchemaArray\n")
 			}
 		} else {
-			buf.WriteString("*kallax.BaseSchemaField\n")
+			if !isRootField(field) && field.IsJSON {
```
Comment (AnthonyYatsenko) [inline generator/template.go:296]: Comparing your fix with solution for slices...
Does it make sense to apply this fix only for pointers to structs only?
P.S. Similar code may be found at line 386, at `genSubschemaInit`. Does it make sense to apply the fix there as well?

Comment (denis90) [reply generator/template.go:296]: I need this function result to match `genSubschemaInit` and everything called before `genSubschemaInit`.
Added a test with struct pointer to ensure it works correctly. https://github.com/veertech/go-kallax/pull/1/commits/690d8601f189a56ca959128ed26bc72224885064

**Review by sfate — APPROVED** [review]
LGTM 👍

**Review by AnthonyYatsenko — APPROVED** [review]
LGTM

**Pull request merged**